internal/chezmoi/sourcestate.go

```diff
@@ -769,6 +770,11 @@ func (s *SourceState) ExecuteTemplateData(options ExecuteTemplateDataOptions) ([
 	templateData := s.TemplateData()
 	if chezmoiTemplateData, ok := templateData["chezmoi"].(map[string]any); ok {
 		chezmoiTemplateData["sourceFile"] = options.Name
+
+		if options.Destination != "" {
+			chezmoiTemplateData["targetDir"] = filepath.Dir(options.Destination)
+			chezmoiTemplateData["targetFile"] = filepath.Base(options.Destination)
```

Comment:
I'd prefer to keep the previous behavior of having `targetFile` be the full path to the target file. The user can strip the destination prefix if they want.

Reply:
I will make those changes (easy enough).

Because I changed an existing test, I am going to split that test so that the original test is not lost.